src/app/parcels/ExpandedParcelDetails.tsx

```diff
@@ -11,20 +17,56 @@ const sortByTimestampWithMostRecentFirst = (events: EventTableRow[]): EventTable
     return events.sort((eventA, eventB) => eventB.timestamp.getTime() - eventA.timestamp.getTime());
 };
 
-const ExpandedParcelDetails = async ({ parcelId }: Props): Promise<React.ReactElement> => {
-    if (!parcelId) {
-        return <></>;
+function getErrorMessageForExpandedParcelDetailsError(
+    error: FetchExpandedParcelDetailsError
+): string {
+    switch (error.type) {
+        case "failedToFetchParcelDetails":
+            return `Failed to fetch parcel details. Log ID: ${error.logId}`;
+        case "clientDetailDoesNotExist":
+            return `Client detail cannot be found. Log ID: ${error.logId}`;
     }
-    const expandedParcelDetails = await getExpandedParcelDetails(parcelId);
+}
+
+const ExpandedParcelDetailsView = ({ parcelId }: Props): ReactElement => {
+    const [parcelDetails, setParcelDetails] = useState<ExpandedParcelDetails | null>(null);
+    const [errorMessage, setErrorMessage] = useState<string | null>(null);
+
+    const fetchAndSetParcelDetails = useCallback(async (): Promise<void> => {
+        if (!parcelId) {
+            return;
+        }
+
+        const { parcelDetails: expandedParcelDetails, error } =
+            await getExpandedParcelDetails(parcelId);
+
+        if (error) {
+            const newErrorMessage = getErrorMessageForExpandedParcelDetailsError(error);
+            setErrorMessage(newErrorMessage);
+            return;
+        }
+
+        setParcelDetails(expandedParcelDetails);
+    }, [parcelId]);
+
+    useEffect(() => {
+        void fetchAndSetParcelDetails();
+    }, [fetchAndSetParcelDetails]);
 
     return (
         <>
-            <DataViewer data={expandedParcelDetails.expandedParcelData} />
-            <EventTable
-                tableData={sortByTimestampWithMostRecentFirst(expandedParcelDetails.events)}
-            />
+            {errorMessage && <ErrorLargeText>{errorMessage}</ErrorLargeText>}
```

Comment:
just to point out I was using `<ErrorSecondaryText>` in the action modals - should we choose one to be consistent? Don't mind which

Reply:
happy to use ErrorSecondaryText! changed now.